File: Lista nr 3/Hexagon.cpp
#include "Hexagon.hpp"
#define Constant 2.59808
#include "NegativeNumber.hpp"
Hexagon::Hexagon(int a)
{
    if(a < 0)
        throw NegativeNumber();
    side = a;
}

int Hexagon::area()
{
    return side * side * Constant;
}
int Hexagon::perimeter()
{
    return 6 * side;
}
std::string Hexagon::getNAME()
{
    return "Hexagon";
}

Hexagon::~Hexagon()
{

}

File: Lista nr 3 Java/Hexagon.java
class Hexagon implements Figure
{
    private static final double Constant = 2.59808;
    private int side;
    public Hexagon(int a) throws NegativeNumber
    {
        if(a < 0)
        throw new NegativeNumber();
        side = a;
    }
    public int area()
    {
        return (int)(side * side * Constant);
    }
    public int perimeter()
    {
        return 6 * side;
    }
    public String getNAME()
    {
        return "Hexagon";
    }

}